backend/api/quivr_api/modules/knowledge/repository/knowledges.py

```diff
@@ -35,6 +35,13 @@ async def insert_knowledge(self, knowledge: KnowledgeDB) -> KnowledgeDB:
             if existing_knowledge:
```

Comment:
This check shouldn't be on file_name an brain_id. (in later PR)
We should probably take an `upsert: bool` arg to upsert the knowledge if it exists and upsert is True

Reply:
the check is done before hand, maybe not check twice ? though, if the file_name exists maybe suggest a new name ?